bin/compile

```diff
@@ -1,28 +1,95 @@
 #!/bin/bash
 
+set -e
+
 BUILD_DIR=$1
-#CACHE_DIR=$2
+CACHE_DIR=$2
 ENV_DIR=$3
 BUILDPACK_DIR="$(dirname "$(dirname "$0")")"
 
+function main() {
+  export_env_dir "$ENV_DIR"
+
+  # Get latest stable version from the tailscale pkgs page
+  TAILSCALE_VERSION=${TAILSCALE_VERSION:-$(curl --silent https://pkgs.tailscale.com/stable/ | \
+    awk '/option/ {gsub("<[^>]+>", ""); print $1}' | \
+    grep -v latest | sort -r | head -1)}
+
+  # if we have a cached copy of the tailscale executable...
+  if [ -f "$CACHE_DIR/tailscale" ]; then
+    cached_version=$("$CACHE_DIR/tailscale" --version | head -1)
+
+    # ...and the cached version matches the requested version.
+    if [ "$cached_version" = "$TAILSCALE_VERSION" ]; then
+      echo "Installing Tailscale v$TAILSCALE_VERSION from cache" | major
+      copy_binaries_from_cache
+    fi
+  fi
+
+  if [ ! -f "$BUILD_DIR/bin/tailscale" ]; then
+    echo "Installing Tailscale v$TAILSCALE_VERSION" | major
+    install_from_apt
+    copy_binaries_to_cache
+    copy_binaries_from_cache
+  fi
+
+  copy_start_script
+  echo "Tailscale installed" | major
+  echo
+}
+
 function indent() {
   sed -u 's/^/       /'
 }
 
-if [[ ! -e "$ENV_DIR/TAILSCALE_AUTHKEY" ]]; then
```

Comment:
Removed this check as we don't actually care at build time if you've got the TAILSCALE_AUTHKEY variable set. That only matters at run time. I was using this initially as a poor man's "env check" thinking that it would be better for this to fail at build time than at run time. But upon reflection, in situations like ours where we have a separate build environment, this also means we need to set the AUTHKEY in the build environment in addition to our production environment, which feels off when you take into consideration that this build script doesn't do anything with that variable at all.